src/opers.h

```diff
@@ -386,41 +416,53 @@ static inline UInt NRB_FLAGS(Obj flags)
 **  is either 'true' or 'false'.  <pos> must  be a positive integer less than
 **  or equal to the length of <hdList>.
 **
-**  Note that 'ELM_FLAGS' is a macro, so do not call it  with arguments  that
-**  have side effects.
 **
 **  'C_ELM_FLAGS' returns a result which it is better to use inside the kernel
 **  since the C compiler can't know that True != False. Using C_ELM_FLAGS
 **  gives slightly nicer C code and potential for a little more optimisation.
 */
-#define C_ELM_FLAGS(list, pos)                                               \
-    ((BLOCK_ELM_FLAGS(list, pos) & MASK_POS_FLAGS(pos)) ? 1 : 0)
+static inline Int C_ELM_FLAGS(Obj list, UInt pos)
+{
+    if (BLOCK_ELM_FLAGS(list, pos) & MASK_POS_FLAGS(pos)) {
+        return 1;
+    }
+    else {
+        return 0;
+    }
+}
 
-#define ELM_FLAGS(list, pos) (C_ELM_FLAGS(list, pos) ? True : False)
+static inline Obj ELM_FLAGS(Obj list, UInt pos)
+{
+    return C_ELM_FLAGS(list, pos) ? True : False;
+}
 
 static inline Int SAFE_C_ELM_FLAGS(Obj flags, UInt pos)
 {
     return (pos <= LEN_FLAGS(flags)) ? C_ELM_FLAGS(flags, pos) : 0;
 }
 
-#define SAFE_ELM_FLAGS(list, pos) (SAFE_C_ELM_FLAGS(list, pos) ? True : False)
+static inline Obj SAFE_ELM_FLAGS(Obj list, UInt pos)
+{
+    return SAFE_C_ELM_FLAGS(list, pos) ? True : False;
+}
```

Comment:
I would suggest to get rid of `SAFE_ELM_FLAGS` and `ELM_FLAGS`; and instead rename `C_ELM_FLAGS` to `ELM_FLAGS`, and `SAFE_C_ELM_FLAGS` to `SAFE_ELM_FLAGS`.  Then there is no need for including `bool.h`. And (hypothetical) external code accessing a flags list usually will be better of getting 0/1 return values anyway.

If you agree with this idea, I can also implement it and push that as an addition to this PR.

Reply:
I don't mind it in principle, but there are quite a few places using ELM_FLAGS and SAFE_ELM_FLAGS, and I worry they are going to all look messy if we start explictly converting 0/1 to True/False.

Reply:
There are exactly three each :)

Reply:
Here is the full patch eliminating all uses of them. As I said, I could adjust the PR to this end, or I can make another PR where I make the changes (but that would then of course conflict with this PR, albeit in a minor way.
```patch
diff --git a/src/opers.c b/src/opers.c
index b3d23fdfb..612ab8a9b 100644
--- a/src/opers.c
+++ b/src/opers.c
@@ -1082,7 +1082,7 @@ Obj DoSetFilter (
     flags = FLAGS_TYPE( type );
     
     /* return the value of the feature                                     */
-    if ( val != SAFE_ELM_FLAGS( flags, flag1 ) ) {
+    if ( (val == True) != SAFE_C_ELM_FLAGS( flags, flag1 ) ) {
         ErrorReturnVoid(
             "value feature is already set the other way",
             0L, 0L,
@@ -1125,10 +1125,7 @@ Obj DoFilter (
     flags = FLAGS_TYPE( type );
     
     /* return the value of the feature                                     */
-    val = SAFE_ELM_FLAGS( flags, flag1 );
-    
-    /* return the value                                                    */
-    return val;
+    return SAFE_C_ELM_FLAGS( flags, flag1 ) ? True : False;
 }
 
 
@@ -2478,7 +2475,7 @@ Obj DoTestAttribute (
     flags = FLAGS_TYPE( type );
 
     /* return whether the value of the attribute is already known          */
-    return SAFE_ELM_FLAGS( flags, flag2 );
+    return SAFE_C_ELM_FLAGS( flags, flag2 ) ? True : False;
 }
 
 
@@ -2865,7 +2862,7 @@ Obj DoSetProperty (
 
     /* if the value of the property is already known, compare it           */
     if ( SAFE_C_ELM_FLAGS( flags, flag2 ) ) {
-        if ( val == ELM_FLAGS( flags, flag1 ) ) {
+        if ( (val == True) == C_ELM_FLAGS( flags, flag1 ) ) {
             return 0;
         }
         else {
@@ -2933,7 +2930,7 @@ Obj DoProperty (
 
     /* if the value of the property is already known, simply return it     */
     if ( SAFE_C_ELM_FLAGS( flags, flag2 ) ) {
-        return ELM_FLAGS( flags, flag1 );
+        return C_ELM_FLAGS( flags, flag1 ) ? True : False;
     }
 
     /* call the operation to compute the value                             */
@@ -2989,7 +2986,7 @@ Obj DoVerboseProperty (
 
     /* if the value of the property is already known, simply return it     */
     if ( SAFE_C_ELM_FLAGS( flags, flag2 ) ) {
-        return ELM_FLAGS( flags, flag1 );
+        return C_ELM_FLAGS( flags, flag1 ) ? True : False;
     }
 
     /* call the operation to compute the value                             */
```

Reply:
I haven't done this, I'll be honest, I think this makes the code quite a bit harder to read.

Reply:
Note that we use this pattern (`val ? True : False` already in countless places... I don't share the view that it's hard to read, though I guess one could come up with a macro/helper function to hide it away... But frankly, I'd then probably find *that* harder to read, as to understand that, one would have to lookup what that macros does.

Anyway, I can make a PR with the changes I have in mind separately, and we can discuss it further there (or not)